tests/conftest.py

```diff
@@ -72,12 +76,77 @@ def firefox_notifications(notifications):
     params=[DESKTOP],
     ids=['Desktop'],
 )
-def selenium(selenium, request):
-    """Fixture to set a custom resolution for tests running on Desktop."""
+def selenium(selenium, base_url, session_auth, request):
+    """Fixture to set a custom resolution for tests running on Desktop
+    and handle browser sessions when needed"""
     selenium.set_window_size(*request.param)
+    # establishing actions  based on markers
+    create_session = request.node.get_closest_marker('create_session')
+    login = request.node.get_closest_marker('login')
+    clear_session = request.node.get_closest_marker('clear_session')
+    # this is used when we want to open an AMO page with a sessionid
+    # cookie (i.e. a logged-in user) already set
+    if create_session:
+        # need to set the url context if we want to apply a cookie
+        # in order to avoid InvalidCookieDomainException error
+        selenium.get(base_url)
+        # set the sessionid cookie
+        selenium.add_cookie(
+            {
+                "name": "sessionid",
+                "value": session_auth,
+            }
+        )
+    # this is used when we want to start the browser with a normal login
+    # mostly used for the scope of getting the session cookie and storing it for later use
+    if login:
+        home = Home(selenium, base_url).open().wait_for_page_to_load()
+        home.header.click_login()
+        home.wait.until(
+            EC.visibility_of_element_located((By.NAME, 'email')),
+            message=f'FxA email input field was not displayed in {selenium.current_url}',
+        )
+        user = login.args[0]
+        Login(selenium, base_url).account(user)
+        home.wait.until(
+            EC.url_contains('addons'),
+            message=f'AMO could not be loaded in {selenium.current_url}',
+        )
+        session_cookie = selenium.get_cookie('sessionid')
+        with open(user + '.txt', 'w') as file:
+            file.write(session_cookie['value'])
```

Comment:
Don't forget [this](https://docs.pytest.org/en/6.2.x/tmpdir.html)

Reply:
Yes, as I mentioned in my previous comment, I wasn't very happy with how the `temdir (temp_path)` fixture handles files, for this scope at least. Here is the whole reasoning:

> I've also looked into the `tmpdir` fixture, I've tried to use it to see how it works, but I wasn't very happy about the way it handles files, at least for my purpose. More specific, the files created by `tmpdir` are rewritten whenever a new test session starts, but they are not automatically deleted when a session ends, which means that the data stored in the file will continue to exist until a new session is started and overwrites it - and so on and so forth.
In my case, I want to make sure that any file created during a test session is completely removed when the session ends, hence I kept using my previous method for creating and deleting files, but now everything is incorporated in the selenium fixture.

Reply:
> new test session

Can you be specific here? Are you saying an individual test or the entire suite of tests.

Reply:
I'm talking about an entire suite - because I need the file to exist for the duration of the test suite in order to reuse the session cookie. The file will be created when we call the selenium fixture with the `@pytest.mark.login` marker. Once the entire test session is competed, the temp file stays there (not automatically deleted). The file will be overwritten when I'm starting the test suite again, but it never disappears completely.

Reply:
Are you sure it doesn't remove the dir and file?? That is the whole point of that fixture. I know sometimes it doesn't remove it right away but it should be named differently each time

Reply:
So, if I'm using the fixture without the `--basetemp` argument, the dir and files are created in the system Temp folder and, as you said, the dir is renamed differently each time the fixture is called.  However, the dirs are not deleted as far as I can tell, but there will be only 4  (3 from history and one current) dirs at any given time, so at least they don't stack indefinitely; As for the files, they will still be available for at least 24h after creation time - I have files created from yesterday's run and they are still there at the time of writing this. 
If I'm using the `--basetemp` argument, the behavior is almost the same, the difference being that I can specify a location where the dirs should be created and there will be only one dir left in history (instead of 3, when I'm not using the argument).

Reply:
Hey Lexa. Willing to give your config a try. I will have more time next week to look at it but I know you'd like to try testing some of this. I still would like you to use the ```tmp_path``` fixture but maybe we can talk about that later.

Reply:
Thanks Benny. I would really like to start using this live because I have written a bunch of API tests that rely on this config, so I'd like to be able to push them soon.